### DISCUSSION TITLE
WebSocket + TLS + Web + Cloudflare问题

### DISCUSSION THREAD

**Discussion by maoxikun** [v2fly/v2ray-core#2522] [Q&A] [answered]
+ 使用阿里云dns解析域名参考白话文教程的WebSocket + TLS + Web成功出国，现在想把dns解切换到cf再套一层cdn，步骤如下：
1、把nginx里面的证书切换到cf的源服务器证书
2、cf设置dns解析为代理模式
3、cf设置SSL/TLS 加密模式为 完全（严格）
+ 结果连不上，v2客户端日志如下：
2023/05/18 12:42:54 [Warning] [3274690013] app/proxyman/outbound: failed to process outbound traffic > proxy/vmess/outbound: failed to find an available destination > common/retry: [transport/internet/websocket: failed to dial WebSocket > transport/internet/websocket: failed to dial to (wss://xxxx.xxxx.com:10000/v2ray):  > transport/internet/websocket: dial TLS connection failed > dial tcp [2606:4700:3030::ac43:b6ea]:10000: i/o timeout transport/internet/websocket: failed to dial WebSocket > transport/internet/websocket: failed to dial to (wss://xxxx.xxxx.com:1111/v2ray):  > transport/internet/websocket: dial TLS connection failed > dial tcp: lookup xxxx.xxxx.com: operation was canceled] > common/retry: all retry attempts failed
2023/05/18 12:43:25 [Warning] [3212394295] app/proxyman/outbound: failed to process outbound traffic > proxy/vmess/outbound: failed to find an available destination > common/retry: [transport/internet/websocket: failed to dial WebSocket > transport/internet/websocket: failed to dial to (wss://xxxx.xxxx.com:11111/v2ray):  > transport/internet/websocket: dial TLS connection failed > dial tcp [2606:4700:3030::ac43:b6ea]:10000: i/o timeout transport/internet/websocket: failed to dial WebSocket > transport/internet/websocket: failed to dial to (wss://xxxx.xxxx.com:10000/v2ray):  > transport/internet/websocket: dial TLS connection failed > dial tcp: lookup xxxx.xxxx.com: operation was canceled] > common/retry: all retry attempts failed
+ 有大佬能解答下吗，或者给个教程也行啊

**Comment by qianbinbin** [comment]
其它和你差不多，我用的是 Let's Encrypt 证书
1. 源服务器证书是什么，用它搭一个正常网站能访问吗
2. 确认能直接访问 CDN 吗
3. 同样的网络环境，我用 macOS 没有问题，但有几台 Linux 设备断断续续，后来走纯 IPv4 就好了，原因不明，你不妨试试

**Reply by maoxikun** [reply]
> 其它和你差不多，我用的是 Let's Encrypt 证书
> 
> 1. 源服务器证书是什么，用它搭一个正常网站能访问吗
> 2. 确认能直接访问 CDN 吗
> 3. 同样的网络环境，我用 macOS 没有问题，但有几台 Linux 设备断断续续，后来走纯 IPv4 就好了，原因不明，你不妨试试

cf下载的证书，nginx正常访问，vps是纯ipv4

**Comment by nursery01** [comment, marked as answer]
ws使用了cloudflare的10000埠，肯定連不上

cloudflare支援以下埠
```
443
2053
2083
2087
2096
8443
```

[https://developers.cloudflare.com/fundamentals/get-started/reference/network-ports/](https://developers.cloudflare.com/fundamentals/get-started/reference/network-ports/)

**Reply by qianbinbin** [reply]
我靠，，感觉是破案了😄

**Reply by maoxikun** [reply]
> ws使用了cloudflare的10000埠，肯定連不上
> 
> cloudflare支援以下埠
> 
> ```
> 443
> 2053
> 2083
> 2087
> 2096
> 8443
> ```
> 
> https://developers.cloudflare.com/fundamentals/get-started/reference/network-ports/

感谢，没注意cf还有这个限制，有时间试试，现在把dns转到dnspod了
